crates/core/tedge_api/src/parser.rs

```diff
@@ -72,6 +78,119 @@ struct ThinEdgeValueParser<'key, 'vis, T> {
     visitor: &'vis mut T,
 }
 
+pub struct IsoOrUnix(OffsetDateTime);
+
+impl IsoOrUnix {
+    pub fn new(offset_date_time: OffsetDateTime) -> Self {
+        Self(offset_date_time)
+    }
+
+    pub fn reformat_value(value: Value, time_format: TimeFormat) -> serde_json::Result<Value> {
+        Ok(match (&value, time_format) {
+            (Value::Number(_), TimeFormat::Unix) => value,
+            (Value::String(_), TimeFormat::Rfc3339) => value,
+            (_, TimeFormat::Unix) => {
+                let Self(time) = serde_json::from_value(value)?;
+                time.unix_timestamp().into()
+            }
+            (_, TimeFormat::Rfc3339) => {
+                let Self(time) = serde_json::from_value(value)?;
+                Value::String(time.format(&Rfc3339).unwrap())
+            }
+        })
+    }
+
+    pub fn format(&self, time_format: TimeFormat) -> serde_json::Value {
+        match time_format {
+            TimeFormat::Unix => Value::Number(self.0.unix_timestamp().into()),
+            TimeFormat::Rfc3339 => Value::String(self.0.format(&Rfc3339).unwrap()),
+        }
+    }
+}
```

Comment:
The reformatting can be part of a method on `TimeFormat`, but the `IsoOrUnix` struct exists to support deserialization for the case of Cumulocity values, which are handled very differently to Azure/AWS values. Because we need this deserialization logic in `tedge_api` (e.g. in `alarm.rs`) and also need the same logic to reformat timestamps for Azure and AWS, somewhere we require a crate that knows about both `IsoOrUnix` and `TimeFormat`. At the moment, this is `tedge_api` depending on `tedge_config` for `TimeFormat`, but this could be reversed, or `tedge_config` could implement the logic above and `tedge_api` can just use the implementation there for deserialization as required.